lib/connection/connections/HttpConnection.ts

```diff
@@ -21,12 +21,13 @@ export default class HttpConnection implements IConnectionProvider, IThriftConne
   private connection: any;
 
   connect(options: IConnectionOptions, authProvider: IAuthentication): Promise<IThriftConnection> {
+    const Agent = options.options?.https === false ? http.Agent : https.Agent;
```

Comment:
One minor suggestion, otherwise looks good:

```suggestion
    const Agent = options.options?.https ? https.Agent : http.Agent;
```

Reply:
I'm happy to do this, but noting that this will default to use the `http.Agent` (not `https.Agent`) if any of `options.options.http` is undefined. Is that ok?